src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1131,6 +1270,27 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::do_init_tm_tx(
     }
     auto term = term_opt.value();
 
+    auto cfg = _metadata_cache.local().get_topic_cfg(model::tx_manager_nt);
+    if (!cfg) {
+        co_return init_tm_tx_reply{tx_errc::partition_not_exists};
+    }
+    if (stm->hosted_transactions_inited()) {
```

Comment:
add this check to other rpc methods (add partition, add offstes and end txn); you may hide hosted_transactions_inited inside hosts and return true if hosted transactions aren't init-ed